src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -65,33 +65,34 @@ void ColumnVsValueTableScanImpl::_scan_segment(const BaseSegment& segment, const
 void ColumnVsValueTableScanImpl::_scan_segment(const BaseDictionarySegment& segment, const ChunkID chunk_id,
                                                PosList& matches,
                                                const std::shared_ptr<const PosList>& position_filter) const {
-  /*
-   * ValueID value_id; // left value id
-   * Variant value; // right value
+  /**
```

Comment:
The issue you linked sounds worrisome, but I am quite sure I haven't touched handling of null values in this PR.

Reply:
I am using INVALID_VALUE_ID as returned by DictionarySegment::lower_bound/upper_bound to indicate "not found" - not in the null-value sense.

Reply:
I closed the linked issue because it was based on a misunderstanding. Still, I have a bad feeling about our NULL handling here. Not necessarily saying that you changed it, but it feels odd.

Reply:
You mean that ` _scan_with_iterators<false>(always_true, it, end, chunk_id, matches);` in the MatchesAll case should probably have "true" as template parameter? I feel the same right now...

Reply:
I'll quickly look into this and PR/issue if necessary. Not really related to this PR

Reply:
Not in all cases. If we are searching for `< existing_value`, we are good. If that `<` became a `>`, we might run into problems. I screwed up my build setup, so it will take me a while to test this.

If this is the case, my question concerning testing is answered - even if not in the way I hoped for. ;)

Reply:
Was this meant as a comment for https://github.com/hyrise/hyrise/pull/1349#discussion_r240342454? 

If that's the case: I breakpointed into both early outs and they were triggered repeatedly by the mentioned tests

Reply:
No, it was meant as "If my guess with the early outs for the given example is right and it does actually fail, the tests are inadequate."